### PR TITLE
Reduced KVO dependency

### PR DESCRIPTION
**Pull request by lickel**
We can use an observation token to abstract a KVO layer.
I've also created an intermediate RawData type.

I wanted to make this generic, but the same compiler bugs we've seen in the past still exist.
I think, given that, my next PR I plan to make something like a CWJSON enum/struct.
I'll also make a point of having it conform to dynamic member lookup so we can do stuff like:

`let organizationID = data.organizationId?.id?.stringValue` or something similar.

Note: I am currently requiring the class to have an obj-c `id` getter.
I _can_ abstract that away in a different fashion if that's preferable.

Basically: one version uses `NSSortDescriptor(key: "id")` the other would use an `NSSortDescriptor("self")` with a closure to compare the IDs.

### CODE REVIEW

File: FetchRequests/Sources/CWFetchableObject.swift [hunk 1]
```diff
@@ -52,31 +33,16 @@ public protocol CWFetchableObjectProtocol: class, CWIdentifiable, CWRawDataRepre
...
-        }
-    }
-}
-
-#if canUseRawData
-extension CWFetchableObjectProtocol where RawData: Equatable {
-    static func rawDataIsIdentical(lhs: RawData, rhs: RawData) -> Bool {
-        return lhs == rhs
-    }
+    public func listenForUpdates() {}
```
Comment (AdamPro13) [inline FetchRequests/Sources/CWFetchableObject.swift]: Should we actually be providing a default implementation for this method? This seems like a method developers should need to be forced to implement

File: FetchRequests/Sources/Controller/CWFetchedResultsController.swift [hunk 2]
```diff
@@ -906,24 +891,27 @@ private extension CWFetchedResultsController {
         }
 
         for (index, sort) in sortDescriptors.enumerated() {
+            guard let keyPath = sort.key, keyPath != "self" else {
```
Comment (AdamPro13) [inline FetchRequests/Sources/Controller/CWFetchedResultsController.swift]: Where does the `self` thing come from?

Comment (lickel) [reply FetchRequests/Sources/Controller/CWFetchedResultsController.swift]: Oh good catch. I had played with some alternatives for responding to "id" that I ripped out but I think I want to put back.

File: FetchRequests/Sources/CWFetchableObject.swift (hunk 1 above)
Comment (lickel) [reply FetchRequests/Sources/CWFetchableObject.swift]: I was guessing that a lot of frameworks would automatically just do this by default?
Like if I had built CWObject initially I wouldn't have required opting into live updates.

File: FetchRequests/Sources/Controller/CWFetchedResultsController.swift (hunk 2 above)
Comment (lickel) [reply FetchRequests/Sources/Controller/CWFetchedResultsController.swift]: Pushed the relevant change

File: FetchRequests/Sources/CWFetchableObject.swift (hunk 1 above)
Comment (AdamPro13) [reply FetchRequests/Sources/CWFetchableObject.swift]: Sounds good to me
